src/browser-runner/platform-electron/sagas/server-command-handlers.js

```diff
@@ -28,6 +28,14 @@ function* createWindow({ meta: client, payload: { winId, url, width, height, sty
     }[style],
   });
 
+  win.on('close', (e) => {
+    // console.log('Window closing!!!', win.webContents);
+    // I'd like to do this only if there are no tabs left.
+    // Any ideas how I could get the number of tabs from here?
+    e.preventDefault();
+    return false;
+  });
```

Comment:
Ok, to the crux of this PR :)

This issue #31 has a link to a diagram with some extremely high level diagram of a proposed browser architecture (I'm sure you've seen it). While there are many open questions still, and nothing there is set in stone, one thing stands out: the exact data flow, and more importantly, where and what data lives where isn't particularly decided at this point. However, current forces steer us to storing "some of the things" in browser frontends for now. Whether or not we end up (or *can* end up) putting all of the application state in a single one of those 3 components (e.g. browser server) remains up to debate (probably not).

However, as it stands right now, the browser runner (where this particular event handler you've written lives) should not make the decision to close the browser window when "no tabs remain", since runners have no concept of "frontend" tabs at the moment. 

What should happen instead is, similarly to what happens when you click the close window button on Windows and Linux, an action is sent to the browser server "requesting" [1] that a window is closed (which is itself received and handled here [2]). This, in turn, will instruct the browser runner (platform) to close the window [3], and if no more windows are available [4], quit the application if necessary [5]. All of this architecture is in place to avoid switching between electron, qbrt, some future servo-based runner etc., all while allowing these runners to be very easily written (as they make no assumptions about how things should work, and only respond to commands).

Back to the problem at hand, the "correct" approach here is to just send this `SharedActions.events.fromFrontend.toServer.app.window.requestedClose()` action to the browser server when no more tabs are open (and this check should be made in the frontend).

[1] https://github.com/victorporof/tofino/blob/bfg-browser/src/browser-frontend/views/browser/chrome/decorations/window-controls.jsx#L34
[2] https://github.com/victorporof/tofino/blob/bfg-browser/src/browser-server/sagas/frontend-event-handlers.js#L25
[3] https://github.com/victorporof/tofino/blob/bfg-browser/src/browser-runner/platform-electron/sagas/server-command-handlers.js#L42
[4] https://github.com/victorporof/tofino/blob/bfg-browser/src/browser-server/sagas/runner-event-handlers.js#L79
[5] https://github.com/victorporof/tofino/blob/bfg-browser/src/browser-runner/platform-electron/sagas/server-command-handlers.js#L81